clippy_lints/src/literal_representation.rs

```diff
@@ -203,6 +203,11 @@ impl<'a> DigitInfo<'a> {
                 .rev()
                 .collect::<Vec<String>>()
                 .join("_");
+            // Forces hexadecimal values to be grouped by 4 being filled with zeroes (e.g 0x00ab_cdef)
+            let nb_digits_to_fill = self.digits.chars().filter(|&c| c != '_').collect::<Vec<_>>().len() % 4;
```

Comment:
Since this is almost doing the same thing as what's happening above, can you store the intermediate `Vec` from above in a variable and reuse it here? The reversion above is irrelevant for counting.